src/stream/mod.rs

```diff
@@ -87,6 +87,15 @@ impl Stream {
             }
         }
 
+        if let Err(reason) =
```

Comment:
For the future reference, we should think how stable is this and add a command line argument to enable and disable this functionality.

Reply:
true